#! /usr/bin/env python
# Create a cfg file for getting control parameters for dmoc optimal controller

PACKAGE = 'gcop_ctrl'

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

#       Name          				Type      Reconfiguration level        Description   				Default   			Min   			 Max
#gen.add("iterate"     ,     bool_t     ,     0     ,     "Run one iteration of dmoc"  ,     False )
gen.add("usemocap"        , bool_t   , 0                 , "To use mocap or not"          , False)
gen.add("Nit"             , int_t    , 0                 , "Number of iterations of dmoc" , 30     , 0   , 30)
gen.add("reinitialize"    , int_t    , 0                 , "Reinitialize controls"        , 0     , -1   , 1)
gen.add("tf"              , double_t , 0                 , "Final Time"                   , 5      , 0   , 10	)
gen.add("N"               , int_t    , 0                 , "Number of segments"           , 64     , 0   , 100	)
gen.add("x0"              , double_t , 0                 , "Initial xposn"                , -1.0   , -10 , 10	)
gen.add("y0"              , double_t , 0                 , "Initial yposn"                , -1.0   , -10 , 10	)
gen.add("theta0"          , double_t , 0                 , "Initial car angle"            , 1.0    , -5  , 5	)
gen.add("v0"              , double_t , 0                 , "Initial yvel"                 , 0      , -5  , 5	)
gen.add("xN"              , double_t , 0                 , "Final xposn"                  , 0      , -10 , 10	)
gen.add("yN"              , double_t , 0                 , "Final yposn"                  , 0      , -10 , 10	)
gen.add("thetaN"          , double_t , 0                 , "Final angle of the car"       , 0      , -5  , 5	)
gen.add("vN"              , double_t , 0                 , "FinalBody ang vel"            , 0      , -5  , 5	)
gen.add("Q1"              , double_t , 0                 , "Statecost"                    , 0      , 0   , 100	)
gen.add("Q2"              , double_t , 0                 , "Statecost"                    , 0      , 0   , 100	)
gen.add("Q3"              , double_t , 0                 , "Statecost"                    , 0      , 0   , 100	)
gen.add("Q4"              , double_t , 0                 , "Statecost"                    , 1.0    , 0   , 100	)
gen.add("Qf1"             , double_t , 0                 , "FinStatecost"                 , 50     , 0   , 100	)
gen.add("Qf2"             , double_t , 0                 , "FinStatecost"                 , 50     , 0   , 100	)
gen.add("Qf3"             , double_t , 0                 , "FinStatecost"                 , 30     , 0   , 100	)
gen.add("Qf4"             , double_t , 0                 , "FinStatecost"                 , 1      , 0   , 100	)
gen.add("R1"              , double_t , 0                 , "Controlcost"                  , 1      , 0   , 100	)
gen.add("R2"              , double_t , 0                 , "Controlcost"                  , 0.1    , 0   , 100	)
gen.add("mu"              , double_t , 0                 , ""                             , 0.01   , 0   , 100	)

exit(gen.generate(PACKAGE , "Config" , "DMocInterface"))




